### PR TITLE
W-11203142: Descriptors generation API: include test dependencies and resources when needed

### PR DESCRIPTION
**Pull request by rbourbonnavarro**

### CODE REVIEW

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java [hunk 1]
```diff
@@ -111,8 +107,49 @@ protected static MavenConfiguration getDefaultMavenConfiguration() {
     return mavenConfigurationBuilder.build();
   }
 
-  protected AbstractMavenDeployableProjectModelBuilder(MavenConfiguration mavenConfiguration) {
+  protected AbstractMavenDeployableProjectModelBuilder(MavenConfiguration mavenConfiguration, File projectFolder) {
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:110]: wouldn't this break places where this is already extended?

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java [hunk 2]
```diff
@@ -111,8 +107,49 @@ protected static MavenConfiguration getDefaultMavenConfiguration() {
...
     this.mavenConfiguration = mavenConfiguration;
+    this.projectFolder = projectFolder;
+  }
+
+  @Override
+  public DeployableProjectModel build() {
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:116]: how is this relevant to the feature being added?

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java [hunk 3]
```diff
@@ -111,8 +107,49 @@ protected static MavenConfiguration getDefaultMavenConfiguration() {
...
+      return emptyList();
+    }
+
+    final Object attributeObject = attributes.getOrDefault(attribute, emptyList());
+    checkArgument(attributeObject instanceof List, format("The '%s' attribute must be of '%s', found '%s'", attribute,
+                                                          List.class.getName(), attributeObject.getClass().getName()));
+    return (List<String>) attributeObject;
+  }
+
+  protected <T> T getSimpleAttribute(Map<String, Object> attributes, String attribute, T defaultValue) {
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:151]: is this ok to be protected?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java (hunk 3 above)
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:151]: make final

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java [hunk 4]
```diff
@@ -111,8 +107,49 @@ protected static MavenConfiguration getDefaultMavenConfiguration() {
...
+    resolveAdditionalPluginDependencies(aetherMavenClient, pomModel, activeProfiles, pluginsArtifactDependencies);
+
+    return doBuild(pomModel, deployableArtifactCoordinates);
+  }
+
+  protected abstract DeployableProjectModel doBuild(Model pomModel, ArtifactCoordinates deployableArtifactCoordinates);
+
+  protected abstract File getPomFromFolder(File projectFolder);
+
+  protected List<String> getAttribute(Map<String, Object> attributes, String attribute) {
```
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:140]: make final

File: modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java (hunk 2 above)
Comment (elrodro83) [inline modules/artifact-activation/src/main/java/org/mule/runtime/module/artifact/activation/internal/maven/AbstractMavenDeployableProjectModelBuilder.java:116]: make final?

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Comment by rbourbonnavarro** [comment]
--validate

**Pull request merged**